### PR TITLE
libbpf: Added AttachXDP wrapper

### PR DESCRIPTION
**Pull request by guyarb**
https://github.com/aquasecurity/libbpfgo/issues/171

### CODE REVIEW

**Comment by grantseltzer** [comment]
Hi @guyarb! My apologies for the delay in review, I was off all of last week. I'll be looking at this today.

File: libbpfgo.go [hunk 1]
```diff
@@ -1263,6 +1263,26 @@ func (p *BPFProg) SetAttachType(attachType BPFAttachType) {
...
+	}
+	link := C.bpf_program__attach_xdp(p.prog, C.int(iface.Index))
+	if C.IS_ERR_OR_NULL(unsafe.Pointer(link)) {
+		return nil, errptrError(unsafe.Pointer(link), "failed to attach xdp on device %s to program %s", deviceName, p.name)
+	}
+
+	bpfLink := &BPFLink{
+		link:      link,
+		prog:      p,
+		linkType:  Tracepoint,
```
Comment (grantseltzer) [inline libbpfgo.go:1279]: Are we sure that this is the correct link type?

Comment (guyarb) [reply libbpfgo.go:1279]: great catch! Added new link type `XDP`

**Comment by guyarb** [comment]
> I manually tested this and of course it works great, thanks for submitting! Just a comment and a question. I'd like to have a selftest for this but because it'll require an external program it can get complicated, not going to ask that of you.

I copied the selftest from `tc` and modified it accordingly
Ran it locally and it worked
If I made anything wrong in the selftest - please let me know!

File: selftest/xdp/main.bpf.c [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    // Reserve space on the ringbuffer for the sample
+    process = bpf_ringbuf_reserve(&events, sizeof(int), ringbuffer_flags);
+    if (!process) {
+        return 0;
+    }
+
+    *process = 2021;
+
+    bpf_ringbuf_submit(process, ringbuffer_flags);
+    return 0;
```
Comment (grantseltzer) [inline selftest/xdp/main.bpf.c:30]: Keeping on theme lol

```suggestion
    return XDP_PASS;
```

**Review by grantseltzer — APPROVED** [review]

**Comment by grantseltzer** [comment]
Thank you @guyarb !

File: selftest/xdp/main.bpf.c (hunk 2 above)
Comment (guyarb) [reply selftest/xdp/main.bpf.c]: Oops! Thanks

**Pull request merged**